### ISSUE TITLE
Add a JSON preatty print option to the web api

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#4645]
It would be nice to have an option like:

* pretty: When appending ?pretty to any request made, the JSON returned will be pretty formatted.

**Comment by guyboertje** [comment]
OTOH - you could encourage people to add a [browser extension e.g.](https://chrome.google.com/webstore/detail/json-formatter/bcjindcccaagfpapjjmafapmmgkkhgoa/reviews?hl=en-US)

**Comment by ph** [comment]
I used the extension all the time, but from people using the command line to curl the machine its nice to have and since its only an option to send to the serializer, its easy and useful.

**Comment by guyboertje** [comment]
This from the JrJackson tests:
```ruby

  def test_supports_pretty_printing
    object = {"foo" => 5, "utf8" => "żółć"}

    actual = JrJackson::Json.dump(object, :pretty => true)
    assert_equal "{\n  \"foo\" : 5,\n  \"utf8\" : \"żółć\"\n}", actual
  end
```

**Comment by suyograo** [comment]
> OTOH - you could encourage people to add a browser extension e.g

This is not feasible since a lot of people use curl to debug stuff.

**Comment by purbon** [comment]
working on this now.

**Comment by purbon** [comment]
I agree preatty is good for people using curl

**Comment by purbon** [comment]
Fixed in #4658
